### DISCUSSION TITLE
filter column by multiple values and multiple filterFns

### DISCUSSION THREAD

**Discussion by OrionArm** [TanStack/table#4887] [Q&A]
I'm trying to create an interface in which the user could choose a filtering function (for example, between, more than, etc.), as well as apply several filter values to a column (at least with one, but preferably with different functions).
Is it possible?
I'm trying to recreate the columns and replace the filterFns when the user changes the function in interface, but the filter is not applied
I found some examples for 7 version but nothing for 8
https://codesandbox.io/s/react-table-datagrid-eojw8?file=/src/Components/ReactTable/index.js
https://stackblitz.com/edit/react-ts-qj2sop?file=App.tsx,use-set-filter-type.tsx

**Comment by pi-kei** [comment]
In v8 you can create custom filter function that accepts complex object/array of many filter functions as filterValue and inside that function you can call filter functions according to object/array you provided.

```ts
const filtersMap = {
  between:  (row, columnId, filterValue, addMeta) => { ... },
  moreThan: (row, columnId, filterValue, addMeta) => { ... }
}

type ComplexFilterValue {
  [filterFnName: string]: any
}

const complexFilterFn: FilterFn<any> = function (
  row,
  columnId,
  filterValue,
  addMeta
) {
  const filterConfig = filterValue as ComplexFilterValue
  return Object.entries(filterConfig).every(([filterFnName, filterFnValue]) => filtersMap[filterFnName](row, columnId, filterFnValue, addMeta))
}
```
